css-viewport/Overview.bs

```diff
@@ -260,6 +261,85 @@ as follows:
+
+The <dfn><code>interactive-widgets</code></dfn> property specifies the effect that interactive UI
+widgets have on the page's viewports. It defines whether widgets overlay a given viewport or whether
+the viewport is shrunken so that it remains fully visible while the widget is showing. Interactive
+UI widgets are transient user agent or operating system UI through which a user can provide input.
+
+<div class="note">The most common such UI widget is a virtual keyboard.</div>
+
+The following is a list of valid values for [=interactive-widgets=] and the associated
+viewport-resizing behavior:
+
+<dl>
+	<dt><dfn><code>overlays-content</code></dfn></dt>
+	<dd>
+		Interactive UI widgets MUST NOT [=resize=] the [=initial viewport=] nor
+		the <a spec="CSSOM-VIEW">visual viewport</a>. The user agent must perform the same steps
+		<!--TODO I can't seem to autolink to overlaysContent, help?-->
+		as when <a href="https://www.w3.org/TR/virtual-keyboard/#dom-virtualkeyboard-overlayscontent">
+		VirtualKeyboard.overlaysContent</a> is set to <code>true</code>.
+	</dd>
+	<dt><dfn><code>resize-layout</code></dfn></dt>
+	<dd>
+		Interactive UI widgets MUST [=resize=] the [=initial viewport=] by the interactive widget.
+	</dd>
+	<div class="note">
+		Since the <a spec="CSSOM-VIEW">visual viewport</a>'s size is derived from the
+		[=initial viewport=], [=resize-layout=] will cause a resize of both the initial and visual
+		viewports.
+	</div>
+	<dt><dfn><code>resize-visual</code></dfn></dt>
+	<dd>
+		Interactive UI widgets MUST [=resize=] the <a spec="CSSOM-VIEW">visual viewport</a> but MUST
+		NOT [=resize=] the <a spec="CSS-VIEWPORT">initial viewport</a>.
+	</dd>
+</dl>
+
+If no value, or an invalid value, is set for [=interactive-widgets=], the behavior implied by
+[=resize-visual=] is used as the default.
+
+To <dfn lt="resize">resize a viewport by an interactive widget</dfn>, subtract from it the
+intersection of the viewport rect with the widget's OS reported bounding rect. In cases where this
+would result in a non-rectangular viewport, the behavior is user agent defined.
+
+<div class="note">
+	Some examples where the result would non rectangular: a
+	<a href="https://support.apple.com/en-ca/HT207521">floating or split keyboard</a>, or a keyboard that
+	<a href="https://github.com/MicrosoftEdge/MSEdgeExplainers/blob/main/VirtualKeyboardAPI/explainer.md#motivation">
+	occupies only part</a> of the viewport.
+</div>
+
+<h4 id="interaction-with-virtualkeyboard-overlayscontent">
+	Interaction with virtualKeyboard.overlaysContent
+</h4>
+
+The virtual-keyboard API provides an imperitive API to apply the [=overlays-content=] behavior via
+the <a href="https://www.w3.org/TR/virtual-keyboard/#dom-virtualkeyboard-overlayscontent">
+virtualKeyboard.overlaysContent</a> attribute. This attribute shadows the value set to
+[=interactive-widgets=], namely:
+
+When <code>virtualKeyboard.overlaysContent</code> is set to <code>true</code>, the UA MUST ignore
```

Comment:
This seems like a rather odd thing (that is, that the default for VirtualKeyboard.overlaysContent is different from the default for the meta).

This might be sensible behavior given the current set of things, but maybe it'd be nice to change that, either by changing the `overlaysContent` default or allowing it to be a tri-state or something...

Can you file an issue about that and reference it here perhaps?

Reply:
I'm not sure I understand what you mean by "default". The default value for `overlaysContent` is `false` and the default for the meta is `resize-visual` which is also not `overlays-content`. That seems consistent to me.

They aren't linked, i.e. if you have `<meta name="viewport" content="interactive-widgets=overlays-content">`, `virtualKeyboard.overlaysContent` will still return false unless the author sets it explicitly. Is that what you're referring to?

Reply:
Ah yeah, you're right, I had misunderstood what overlaysContent=true meant. Nvm then. Making the API support the three states may be worth it but also is kind of orthogonal to this.